lib/improver/tests/spotdata/spotdata/test_main.py

```diff
@@ -0,0 +1,196 @@
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the spotdata.main"""
+
+
+import unittest
+import json
+from subprocess import call as Call
+import cf_units
+from tempfile import mkdtemp
+import iris
+from iris.coords import DimCoord
+from iris.tests import IrisTest
+from iris.cube import Cube
+
+from collections import OrderedDict
+import numpy as np
+
+from improver.spotdata.main import run_spotdata as Function
+from improver.spotdata.main import process_diagnostic
+
+
+class Test_main(IrisTest):
+
+    """Test the SpotData framework."""
+
+    def setUp(self):
+        """Create components required for testing framework."""
+        data = np.arange(0, 800, 1)
+
+        data.resize(2, 20, 20)
+        latitudes = np.linspace(-90, 90, 20)
+        longitudes = np.linspace(-180, 180, 20)
+        latitude = DimCoord(latitudes, standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(longitudes, standard_name='longitude',
+                             units='degrees')
+
+        # Use time of 2017-02-17 06:00:00, 07:00:00
+        time = DimCoord(
+            [1487311200, 1487314800], standard_name='time',
+            units=cf_units.Unit('seconds since 1970-01-01 00:00:00',
+                                calendar='gregorian'))
+
+        cube = Cube(data,
+                    long_name="air_temperature",
+                    dim_coords_and_dims=[(time, 0),
+                                         (latitude, 1),
+                                         (longitude, 2)],
+                    units="K")
+
+        orography = Cube(np.ones((20, 20)),
+                         long_name="surface_altitude",
+                         dim_coords_and_dims=[(latitude, 0),
+                                              (longitude, 1)],
+                         units="m")
+
+        self.ancillary_data = {}
+        self.ancillary_data.update({'orography': orography})
+
+        diagnostic_recipe = {
+            "temperature": {
+                "diagnostic_name": "air_temperature",
+                "extrema": False,
+                "filepath": "temperature_at_screen_level",
+                "interpolation_method": "use_nearest",
+                "neighbour_finding": {
+                    "land_constraint": False,
+                    "method": "fast_nearest_neighbour",
+                    "vertical_bias": None
+                    }
+                }
+            }
+
+        self.data_directory = mkdtemp()
+        self.output_directory = mkdtemp()
+
+        self.data_path = (self.data_directory +
+                          '/temperature_at_screen_level.nc')
+        self.orography_path = self.data_directory + '/orography.nc'
+        iris.save(cube, self.data_path)
+        iris.save(orography, self.orography_path)
+
+        self.config_path = self.data_directory + '/spotdata_constants.json'
```

Comment:
It might be worth adding a few inline comments to this `setUp` method to help highlight what you're trying to set up.